File: lab1/SimUDuck/RedheadDuck.cpp
#include "stdafx.h"
#include "RedheadDuck.h"

RedheadDuck::RedheadDuck()
	: Duck(std::make_unique<FlyWithWings>(),
		std::make_unique<QuackBehavior>(),
		std::make_unique<MinuetDanceBehavior>())
{
}

void RedheadDuck::Display() const
{
	std::cout << "I'm redhead duck" << std::endl;
}

File: lw1/src/duck/RedheadDuck.php
<?php

class RedheadDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(new FlyWithWings(), new QuackBehavior(), new MinuetDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m redhead duck!' . PHP_EOL;
    }
}